python-checks/src/test/resources/checks/localVariableAndParameterNameIncompatibility.py

```diff
@@ -65,3 +65,18 @@ def fun10():
     #Noncompliant@+1
     for (builderName, stepName) in foo: #Noncompliant
       pass
+
+
+def type_variables_ok(MyTypeParameter: type):
+    from collections import namedtuple
+    Person = namedtuple('Person', ['name', 'age', 'gender'])
+
+    from typing import NamedTuple
+    Employee = NamedTuple('Employee', [('name', str), ('id', int)])
+
+
+def type_variables_fp():
+    class MyClass:
+        ...
+    MyClassAlias = MyClass  # Noncompliant
```

Comment:
[SONARPY-1662](https://sonarsource.atlassian.net/jira/software/c/projects/SONARPY/issues/SONARPY-1662) tracks this issue. If we have time, we might even want to add it in the sprint.